### PR TITLE
feat(card): create card component

### PR DESCRIPTION
**Pull request by lucamerzi**

### CODE REVIEW

**Comment by lucamerzi** [comment]
Relates #7

**Review by nicobarray — CHANGES_REQUESTED** [review]
In the current state of andive, we only want the simplest card component that defines:
- a white background
- the inner padding
- the border style (no shadow)
- the correct border-radius

(For reference, check the ambler4hospital/trips page cards).

Other features, like foldable content, title or inner dividers will be handled by other components (resp. Accordeon, Typography.H1 and Divider)

I'd love to merge this with the simplest specs. If you want to do a bit more, a customizable root element would be awesome, example:

```
const rootCss = css` // define border & spacings `
const CardRoot = styled.div` ${rootCss} `

function Card({ component, children }) {
  // ? Maybe memoize this RootComponent based on prop.component to
  // ? avoid recreating a new styled instance on each render.
  const RootComponent = styled(component)`${rootCss}` || CardRoot
  // ...
  return <RootComponent>
    {children}
  </RootComponent>
}
```

Feel free to inspire yourself from the above example. This would come handy when using react-grid-system's <Col /> and <Row /> components as `<Card component={Col} xs={12} md={6} />`

**Comment by lucamerzi** [comment]
This PR applies [specs](https://amblerworkspace.slack.com/archives/CHVC3CKN2/p1573650651002300?thread_ts=1573648599.000800&cid=CHVC3CKN2) discussed in the slack channel `product-andive`.

File: docs/stories/card.stories.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import {Card, Info, VSpace} from '@ambler/andive'
+
+const Container = styled.div`
+  width: 50%;
+`
+
+const labelText = 'This is a simple card.'
+
+const itemText =
+  'Cards are non-clickable. They group information together and often have an icon that allows the user to perform actions (edit, add items, etc.)'
```
Comment (nicobarray) [inline docs/stories/card.stories.js]:
```suggestion
  'Cards are non-clickable. They group information together and often have buttons that allow the user to perform actions (edit, add items, etc.)'
```

File: packages/andive/.storybook/config.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ function loadStories() {
   require('../src/stories/app-bar.stories')
   require('../src/stories/autocomplete.stories')
   require('../src/stories/button.stories')
+  require('../src/stories/card.stories')
```
Comment (nicobarray) [inline packages/andive/.storybook/config.js]: This should be in docs/.storybook/config.js

File: packages/andive/src/components/card.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import styled, {css} from 'styled-components'
+import PropTypes from 'prop-types'
+
+import * as palette from '../constants/palette'
+
+const rootCss = css`
+  background: white;
+  padding: 16px;
+  border-radius: 4px;
+  border: 1px solid ${palette.hexToRGBA(palette.lightPrimary)};
```
Comment (nicobarray) [inline packages/andive/src/components/card.js]:
```suggestion
  border: 1px solid ${palette.lightPrimary};
```

palette.hexToRGBA is only used to add transparency to the color.

File: packages/andive/src/components/card.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+function Card({component, children}) {
+  const ComputedCard = React.useMemo(() => computeCard(component), [component])
+  return <ComputedCard>{children}</ComputedCard>
+}
+
+Card.propTypes = {
+  children: PropTypes.node,
+  component: PropTypes.node
+}
```
Comment (nicobarray) [inline packages/andive/src/components/card.js]: I should have said so earlier, but let's make it in typescript (and not use prop-types anymore).

**Review by nicobarray — CHANGES_REQUESTED** [review]
🙌 Thanks for the work, we are close to merging it 👍 Let's just convert it to typescript (the component, not the story).

**Review by nicobarray — APPROVED** [review]
🙌 LGTM. Later we'll ask ourselves how to properly type `children` prop.

**Pull request merged**